### PR TITLE
Mice/Royal Rats can heal off cheese.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

By request, rats and royal rats can now eat cheese wedges, in order to regain small amounts of health. rodents are, after all, ravenous mammalian savages, and should be capable of  eating great deals of cheese, in an ideal world. 

## Why It's Good For The Game

Allows for regal rats to provide an infrequent, but viable source of healing for it's subordinate rats, as well as itself.

## Changelog
:cl:
add: Sentient regal and regular rats can now eat cheese wedges to recover a small amount of health.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 1]
```diff
@@ -236,3 +246,13 @@
 				visible_message("<span class='warning'>[src] chews through the [C]. It looks unharmed!</span>")
 				playsound(src, 'sound/effects/sparks2.ogg', 100, TRUE)
 				C.deconstruct()
+
+/mob/living/simple_animal/hostile/rat/AttackingTarget()
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:250]: what about normal mice?

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Should apply to normal mice

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -95,6 +95,16 @@
 		evolve()
 		return
 
+/mob/living/simple_animal/mouse/ClickOn(atom/A, params)
```
Comment (Time-Green) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:98]: Why use ClickOn? Whoever told you to do that is an idiot.
Anyway, if you really insist on it you should use a proximity check so they can’t eat cheese from 7 tiles away. 

I still recommend you just use AttackingTarget like in corgi code for eating snacks

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 3]
```diff
@@ -95,6 +95,16 @@
...
+	if(istype(A, /obj/item/reagent_containers/food/snacks/cheesewedge) && src.canUseTopic(A, BE_CLOSE, NO_DEXTERITY))
+		if(health == maxHealth)
+			to_chat(src,"<span class='warning'>You don't need to eat or heal.</span>")
+			return
+		to_chat(src,"<span class='green'>You nibble some cheese, restoring your health.</span>")
+		adjustHealth(-(maxHealth-health))
+		qdel(A)
+	..()
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:106]:
```suggestion
	return ..()
```

Comment (Rohesie) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:106]:
```suggestion
		qdel(A)
		return
	return ..()
```

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 4]
```diff
@@ -73,6 +73,16 @@
...
 		. += "<span class='warning'>Who is this foolish false king? This will not stand!</span>"
 
+/mob/living/simple_animal/hostile/regalrat/AttackingTarget()
+	. = ..()
+	if(istype(target, /obj/item/reagent_containers/food/snacks/cheesewedge))
+		if (health >= maxHealth)
+			to_chat(src, "<span class='warning'>You feel fine, no need to eat anything!</span>")
+			return
+		to_chat(src, "<span class='green'>You eat \the [src], restoring some health.</span>")
+		src.heal_bodypart_damage(10)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:83]:
```suggestion
		heal_bodypart_damage(10)
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 5]
```diff
@@ -95,6 +95,16 @@
 		evolve()
 		return
 
+/mob/living/simple_animal/mouse/ClickOn(atom/A, params)
+	if(istype(A, /obj/item/reagent_containers/food/snacks/cheesewedge) && src.canUseTopic(A, BE_CLOSE, NO_DEXTERITY))
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:99]:
```suggestion
	if(istype(A, /obj/item/reagent_containers/food/snacks/cheesewedge) && canUseTopic(A, BE_CLOSE, NO_DEXTERITY))
```

Comment (Rohesie) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:99]: Don't use `src` when it's implied.
Also don't use `canUseTopic`, that's for machine topic. Use `Adjacent()` or the likes.

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 6]
```diff
@@ -236,3 +246,13 @@
...
 				C.deconstruct()
+
+/mob/living/simple_animal/hostile/rat/AttackingTarget()
+	. = ..()
+	if(istype(target, /obj/item/reagent_containers/food/snacks/cheesewedge))
+		if (health >= maxHealth)
+			to_chat(src, "<span class='warning'>You feel fine, no need to eat anything!</span>")
+			return
+		to_chat(src, "<span class='green'>You eat \the [src], restoring some health.</span>")
+		src.heal_bodypart_damage(5)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:257]:
```suggestion
		heal_bodypart_damage(5)
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:98]: Is `ClickOn` still needed with `AttackingTarget` ?

Comment (ArcaneMusic) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:98]: SimpleAnimal's don't have AttackingTarget, that's only on Hostile Simplemobs.

Comment (Rohesie) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:98]: `/mob/living/simple_animal/UnarmedAttack(atom/A, proximity)`
What about this?

**Pull request merged**